test/unit/metric_test.rb

```diff
@@ -246,4 +246,20 @@ def test_parents_does_not_include_duplicates
 
     assert_equal(['parent'], Metric.parents(service_id, %w(child1 child2)))
   end
+
+  def test_descendants
+    service_id = next_id
+    Service.save!(provider_key: 'a_provider_key', id: service_id)
+
+    m1 = Metric.new(service_id: service_id, id: next_id, name: 'metric_level_1')
+    m2 = Metric.new(service_id: service_id, id: next_id, name: 'metric_level_2')
+    m3 = Metric.new(service_id: service_id, id: next_id, name: 'metric_level_3')
+    m1.children = [m2]
+    m2.children = [m3]
+    m1.save
```

Comment:
Same suggestion as in a different commit, we would benefit for generalizing this test to N levels, and perhaps ensure level = 3 works and then some other different random N.

Reply:
See https://github.com/3scale/apisonator/pull/119#discussion_r319086992